src/services/transpile.ts

```diff
@@ -205,7 +205,7 @@ function transpileWorker(input: string, transpileOptions: TranspileOptions, decl
         addRange(/*to*/ diagnostics, /*from*/ program.getOptionsDiagnostics());
     }
     // Emit
-    const result = program.emit(/*targetSourceFile*/ undefined, /*writeFile*/ undefined, /*cancellationToken*/ undefined, /*emitOnlyDtsFiles*/ declaration, transpileOptions.transformers, /*forceDtsEmit*/ declaration);
+    const result = program.emit(/*targetSourceFile*/ undefined, /*writeFile*/ undefined, /*cancellationToken*/ undefined, /*emitOnly*/ undefined, transpileOptions.transformers, /*forceDtsEmit*/ undefined);

```

Comment:
Before we were passing `true` to `forceDtsEmit`, which caused this to evaluate to false:
https://github.com/microsoft/TypeScript/blob/f37482cd16118409284e015409deaf14b416b4a0/src/compiler/emitter.ts#L907
though I'm not sure if this is actually the correct fix. Perhaps it would be to just remove `!forceDtsEmit && ` from `emitter.ts`.

Changing `emitOnlyDtsFiles` to `undefined` actually doesn't seem to matter, at least with the configuration I tested with.

Reply:
Well, no permutation of what I said results in all the tests passing, so clearly not as easy of a fix as I hoped.